src/daft-parquet/src/metadata.rs

```diff
@@ -12,11 +16,220 @@ fn metadata_len(buffer: &[u8], len: usize) -> i32 {
+    type N = ParquetTypeWrapper;
+
+    fn mutate(&mut self, node: Self::N) -> DaftResult<Self::N> {
+        match node.0 {
+            ParquetType::PrimitiveType(mut primitive_type) => {
+                // Fix the `node`'s name
+                let field_id = primitive_type.field_info.id;
+                if let Some(mapped_field) =
+                    field_id.and_then(|field_id| self.field_id_mapping.get(&field_id))
+                {
+                    primitive_type.field_info.name = mapped_field.name.clone();
+                }
+                Ok(ParquetTypeWrapper(ParquetType::PrimitiveType(
+                    primitive_type,
+                )))
+            }
+            ParquetType::GroupType {
+                mut field_info,
+                fields,
+                logical_type,
+                converted_type,
+            } => {
+                // Fix the `node`'s name
+                let field_id = field_info.id;
+                if let Some(mapped_field) =
+                    field_id.and_then(|field_id| self.field_id_mapping.get(&field_id))
+                {
+                    field_info.name = mapped_field.name.clone();
+                }
+
+                let fields = match logical_type {
+                    // GroupTypes with logical types List/Map have intermediate child nested fields without field_ids,
+                    // but we need to keep recursing on them to keep renaming on their children.
+                    Some(_) => fields,
+                    // GroupTypes without logical_types are just structs, and we can go ahead with removing
+                    // any fields that don't have field IDs with a corresponding match in the mapping
+                    None => fields
+                        .into_iter()
+                        .filter(|f| {
+                            f.get_field_info()
+                                .id
+                                .and_then(|field_id| self.field_id_mapping.get(&field_id))
+                                .is_some()
+                        })
+                        .collect(),
+                };
+
+                Ok(ParquetTypeWrapper(ParquetType::GroupType {
+                    field_info,
+                    logical_type,
+                    converted_type,
+                    fields,
+                }))
+            }
+        }
+    }
+}
+
+/// Applies field_ids to a ParquetType, returns `None` if the field ID is not found
+/// in the `field_id_mapping`, or if the field has no field ID.
+fn apply_field_ids_to_parquet_type(
+    parquet_type: ParquetType,
+    field_id_mapping: &BTreeMap<i32, Field>,
+) -> Option<ParquetType> {
+    let field_id = parquet_type.get_field_info().id;
+    if field_id
+        .and_then(|field_id| field_id_mapping.get(&field_id))
+        .is_some()
+    {
+        let mut pq_type_rewriter = ParquetTypeFieldIdRewriter { field_id_mapping };
+        let rewritten_pq_type = ParquetTypeWrapper(parquet_type)
```

Comment:
should be able to call `transform` with a function instead of using `ParquetTypeFieldIdRewriter `